### PR TITLE
Sprk 220

### PR DESCRIPTION
**Pull request by hadarshavit**
Add a CLI option to set autofolio's  timeout and set it by default to 48h (as suggested by Lindauer et al 2015)

### CODE REVIEW

**Comment by thijssnelleman** [comment]
When running the selection example in Sparkle, does that mean it will now take 48h?

**Comment by hadarshavit** [comment]
Yes, we can set it using the new CLI argument to 60 seconds.
In general it is necessary to run AutoFolio for at least a few hours.
In the future we can parallelise it so it will take less time.

**Comment by thijssnelleman** [comment]
> Yes, we can set it using the new CLI argument to 60 seconds. In general it is necessary to run AutoFolio for at least a few hours. In the future we can parallelise it so it will take less time.

That sounds really good.  The 48 hours is far too long as a default value. This should be added as a setting in the sparkle_settings and by default I think a value of 60 seconds is good. Whenever this value is used as a variable, you can use the 48h setting and explain this number with the paper reference in the docstring.

File: CLI/support/compute_marginal_contribution_help.py [hunk 1]
```diff
@@ -303,7 +303,8 @@ def compute_actual_selector_marginal_contribution(
         minimise: bool = True,
         performance_data_csv_path: str = sgh.performance_data_csv_path,
         feature_data_csv_path: str = sgh.feature_data_csv_path,
-        flag_recompute: bool = False) -> list[tuple[str, float]]:
+        flag_recompute: bool = False,
+        selector_timeout=172000) -> list[tuple[str, float]]:
```
Comment (thijssnelleman) [inline CLI/support/compute_marginal_contribution_help.py:307]: Docstring update with paper reference explaining this value

File: CLI/support/construct_portfolio_selector_help.py [hunk 2]
```diff
@@ -117,7 +117,8 @@ def get_selector_fd_id(selector_dir: PurePath) -> int:
 def construct_sparkle_portfolio_selector(selector_path: Path,
                                          performance_data_csv_path: str,
                                          feature_data_csv_path: str,
-                                         flag_recompute: bool = False) -> bool:
+                                         flag_recompute: bool = False,
+                                         selector_timeout=172000) -> bool:
```
Comment (thijssnelleman) [inline CLI/support/construct_portfolio_selector_help.py:121]: Docstring explaining the value

File: sparkle/platform/settings_help.py [hunk 3]
```diff
@@ -78,6 +78,8 @@ class Settings:
     DEFAULT_config_budget_per_run = 600
     DEFAULT_config_number_of_runs = 25
 
+    DEFAULT_portfolio_construction_timeout = 172000
```
Comment (thijssnelleman) [inline sparkle/platform/settings_help.py:81]: Set default value to 60s

File: sparkle/platform/settings_help.py [hunk 4]
```diff
@@ -78,6 +78,8 @@ class Settings:
...
+
```
Comment (thijssnelleman) [inline sparkle/platform/settings_help.py:82]: Also needs a setting parse so user can change this default value from the sparkle_settings.ini

File: tests/test_marginal_contribution_help.py [hunk 5]
```diff
@@ -156,7 +156,8 @@ def test_compute_actual_selector_marginal_contribution(self: TestCase,
             minimise=True,
             performance_data_csv_path=perf_path,
             feature_data_csv_path=feature_csv_path,
-            flag_recompute=True
+            flag_recompute=True,
+            selector_timeout=60
```
Comment (thijssnelleman) [inline tests/test_marginal_contribution_help.py:160]: Nice

**Review by thijssnelleman — CHANGES_REQUESTED** [review]
Needs some changes to explain the ``suggested settings'' from Lindauer et al. (2015) and something that doesn't confuse the user by default (Waiting 48h is unreasonable for an inexperienced user)

**Review by thijssnelleman — APPROVED** [review]

**Pull request merged**